File: frontend/lib/features/auth/domain/entities/user_entity.dart
import 'package:equatable/equatable.dart';

class UserEntity extends Equatable {
  const UserEntity({
    required this.uid,
    required this.name,
    required this.email,
    required this.phone,
    required this.role,
    required this.status,
    this.societyId,
    this.flatId,
    this.flatNumber,
    this.profilePhotoUrl,
    this.createdAt,
    this.updatedAt,
  });

  final String uid;
  final String name;
  final String email;
  final String phone;
  final String role;
  final String status;
  final String? societyId;
  final String? flatId;
  final String? flatNumber;
  final String? profilePhotoUrl;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  bool get isResident => role == 'ROLE_RESIDENT';
  bool get isGuard => role == 'ROLE_GUARD';
  bool get isAdmin => role == 'ROLE_ADMIN';
  bool get isActive => status == 'ACTIVE';

  @override
  List<Object?> get props => [
        uid, name, email, phone, role, status,
        societyId, flatId, flatNumber, profilePhotoUrl,
      ];
}

File: backend/src/main/java/com/myguard/auth/view/UserEntity.java
package com.myguard.auth.view;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;

/**
 * Firestore collection: /users/{uid}
 * Document ID: Firebase UID
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class UserEntity {
    private String uid;
    private String name;
    private String email;
    private String phone;
    private String role;
    private String status;
    private String societyId;
    private String flatId;
    private String flatNumber;
    private String profilePhotoUrl;
    private Instant createdAt;
    private Instant updatedAt;
}
